aggregators/aggregator_test.go

```diff
@@ -648,6 +648,170 @@ func TestAggregateSpanMetrics(t *testing.T) {
 	}
 }
 
+func TestAggregateCombinedMetrics(t *testing.T) {
+	aggIvl := time.Second
+	now := time.Now().Truncate(aggIvl)
+	cmkID := EncodeToCombinedMetricsKeyID(t, "ab01")
+	input := []*TestCombinedMetrics{
+		NewTestCombinedMetrics(
+			WithEventsTotal(1),
+			// Key with very old processing time will be dropped if
+			// it is not within max lookback period.
+			WithKey(CombinedMetricsKey{
+				Interval:       aggIvl,
+				ProcessingTime: now.Add(-time.Hour),
+				ID:             cmkID,
+			}),
+		).AddServiceMetrics(serviceAggregationKey{
+			Timestamp:   now,
+			ServiceName: "test-svc",
+		}).AddTransaction(transactionAggregationKey{
+			TransactionName: "txntestold",
+			TransactionType: "txntypeold",
+		}).AddServiceTransaction(serviceTransactionAggregationKey{
+			TransactionType: "txntypeold",
+		}).GetTest(),
```

Comment:
Maybe simplify this test to be only about lookback? Having all these metrics obscures things IMO.

Reply:
Hmm, good point. I will separate them into 2 sub-tests.

Reply:
Done. Let me know if it still doesn't looks obscure, I can separate it into 2 tests (rather than subtests).

Reply:
Still a little obscure. I think it might help if we did a little refactoring to extract the body of the `t.Run` function into a helper, like this:

```go
func withTestAggregator(t testing.TB, f func(*Aggregator), opts ...Option) (output []*aggregationpb.CombinedMetrics) {
    opts = append([]Option{
        WithDataDir(t.TempDir()),
        WithProcessor(combinedMetricsSliceProcessor(&output)),
        // other test-specific defaults
    }, opts...)

    a, err := New(...)
    require.NoError(t, err)
    defer a.Close(context.Background())

    // Invoke callback to aggregate data.
    f(a)

    err = a.Close(context.Background())
    require.NoError(t, err)
    return output
}
```

That would already be a help. For this specific test, maybe go back to using the same input for both with/without lookback, but rather than encoding the expected slices in either case, evaluate the expectation based on the lookback config. Something like this:

```go
inputs := ...
for _, withLookback := range []bool{false, true} {
    var opts []Option
    if withLookback {
        opts = append(opts, WithLookback(2*time.Hour))
    }
    ...
    output := withTestAggregator(...)
    if withLookback {
        require.NotEmpty(t, output)
        ...
    } else {
        assert.Empty(t, output)
    }
}
```

Just an idea, I'm not 100% sure it's better.

Reply:
(Feel free to merge and iterate on that later.)